tools/mo/openvino/tools/mo/middle/BlockLSTMtoLSTMSequence.py

```diff
@@ -3,13 +3,111 @@
+                               not port.disconnected()]
+        # support only BlockLSTM with one output of concatenated hidden states from all steps
+        if len(connected_out_ports) != 1 or connected_out_ports[0] != 0:
+            return
+        shift_const = block_lstm.forget_bias
+
+        # determine hidden_size based on weights shapes
+        w_shape = block_lstm.in_port(1).data.get_shape()
+        b_shape = block_lstm.in_port(2).data.get_shape()
+        hidden_size = dynamic_dimension
+        if len(b_shape) > 0 and b_shape[0] is not dynamic_dimension:
+            hidden_size = b_shape[0] // 4
+        elif len(w_shape) > 1 and w_shape[1] is not dynamic_dimension:
+            hidden_size = w_shape[1] // 4
+
+        assert hidden_size is not dynamic_dimension
+
+        # normalize weights to the format required by OpenVINO LSTMSequence
+        weights = block_lstm.in_port(1).data.get_value()
+        biases = block_lstm.in_port(2).data.get_value()
+        # 1. reshape weights and bias to highlight channel dimension
+        weights = weights.reshape([weights.shape[0], 4, hidden_size])
+        biases = biases.reshape([4, hidden_size])
+        # 2. reorder gates icfo --> fico for both weights and biases
+        gate_reorder = [2, 0, 1, 3]
+        weights = np.take(weights, gate_reorder, axis=1)
+        biases = np.take(biases, gate_reorder, axis=0)
+        # 3. shift_const.value should be added to the first 1/4th part of the biases (f-gate: 0)
+        # Note: in case of moving this code up before gate reordering, the addition
+        # should be applied at different place
+        biases[0] += shift_const
+        # 4. return to the original shapes
+        weights = weights.reshape([weights.shape[0], -1])
+        biases = biases.flatten()
+        # 5. TF stores weights in IO, but IE requires it in OI: transpose
+        weights = weights.transpose()
+        # 6. set up normalized values for Constant weights and bias
+        block_lstm.in_port(1).data.set_value(weights)
+        block_lstm.in_port(2).data.set_value(biases)
+
+        # re-number h_init_state, c_init_state input ports to match RNNSequence ports order
+        init_hidden_state_source = block_lstm.in_port(3).get_source()
+        init_cell_state_source = block_lstm.in_port(4).get_source()
+        block_lstm.add_output_port(4, skip_if_exist=True)
+        block_lstm.in_port(4).get_connection().set_source(init_hidden_state_source)
+        block_lstm.add_output_port(5, skip_if_exist=True)
+        block_lstm.in_port(5).get_connection().set_source(init_cell_state_source)
+
+        new_attrs = {'sequence_dim': 0,
+                     'batch_dim': 1,
+                     'direction': 'forward',
+                     'hidden_size': hidden_size,
+                     'format': 'tf',
+                     }
+
+        LSTM.update_node_stat(block_lstm, new_attrs)
```

Comment:
That was considered bad practice to change existing node type in the transformation. If I remember correctly one of the concerns against that was that we may have attributes from old node left or inputs/outputs, that are not existing in the new node. We usually create a new node and add all needed attributes.
If I remember correctly `update_node_stat` does create node internally and doesn't inherit attributes from old node, but this is not tested properly so maybe you can add required tests so that this approach can be used.